### PR TITLE
Expanded capabilities in ValidUrl

### PR DESCRIPTION
**Pull request by jace**
* Support a list of allowed schemes and domains
* Optionally disable URL tester
* Use html5lib instead of lxml for parsing safety
* Misc linter fixes

### CODE REVIEW

File: baseframe/forms/form.py [hunk 1]
```diff
@@ -176,7 +176,7 @@ def send_signals(self):
             form_validation_success.send(self)
 
     def errors_with_data(self):
-        # Convert lazy_gettext error strings into unicode so they don't cause problems downstream
+        # Convert lazy_gettext error strings into str so they don't cause problems downstream
```
Comment (jace) [inline baseframe/forms/form.py:179]: Doc fix doesn't belong in this PR. Slipped through.

File: tests/test_validators.py [hunk 2]
```diff
@@ -197,6 +197,250 @@ def test_nonce_form_on_failure(self):
...
+            url = forms.StringField(
+                "URL",
+                validators=[
+                    forms.validators.ValidUrl(allowed_domains=lambda: ('example.net',))
+                ],
+            )
+
+        form = UrlForm(meta={'csrf': False})
+        form.url.data = 'http://www.example.com'
+        assert form.validate() is False
```
Comment (iambibhas) [inline tests/test_validators.py:362]: add a test for `True` as well maybe.

Comment (jace) [reply tests/test_validators.py:362]: That's the next test.
